src/katsdpcontroller/product_config.py

```diff
@@ -2074,7 +2074,7 @@ def _upgrade(config):
         config["version"] = "3.0"
 
     # Upgrade to latest 3.x
-    config["version"] = "3.3"
+    config["version"] = str(schemas.PRODUCT_CONFIG.versions[-1])
```

Comment:
I remember this is what I suggested, but now that I see it I wonder if
```suggestion
    config["version"] = str(max(schemas.PRODUCT_CONFIG.versions))
```
captures the intent better.

Reply:
Does this interpret the strings properly as float numbers? I'm a little hazy on how the parsing works from j2 template to json to python.

Reply:
schemas.PRODUCT_CONFIG.versions are instances of `packaging.version.Version`, which sort the way versions should (see `katsdpcontroller.schemas._normalise_version`).